common/src/main/java/io/quarkus/ts/openshift/common/actions/CopyLogsOnOpenShiftFailureActionImpl.java

```diff
@@ -0,0 +1,39 @@
+package io.quarkus.ts.openshift.common.actions;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.quarkus.ts.openshift.common.Command;
+import io.quarkus.ts.openshift.common.config.Config;
+import io.quarkus.ts.openshift.common.injection.TestResource;
+import io.quarkus.ts.openshift.common.util.OpenShiftUtil;
+
+import java.io.File;
+
+public class CopyLogsOnOpenShiftFailureActionImpl implements OnOpenShiftFailureAction {
+
+    private static final String INSTANCES_LOGS_OUTPUT_DIRECTORY = "instance.logs";
+    private static final String DEFAULT_LOG_OUTPUT_DIRECTORY = "target/logs";
+    private static final String LOG_SUFFIX = ".log";
+
+    @TestResource
+    private OpenShiftUtil openShiftUtil;
+
+    @Override
+    public void execute() throws Exception {
+        String namespace = openShiftUtil.getNamespace();
+        for (Pod pod : openShiftUtil.getPods()) {
+            String podName = pod.getMetadata().getName();
+            new Command("oc", "logs", podName).outputToFile(getOutputFile(podName, namespace)).runAndWait();
+        }
+    }
+
+    static String getOutputFolder() {
+        return Config.get().getAsString(INSTANCES_LOGS_OUTPUT_DIRECTORY, DEFAULT_LOG_OUTPUT_DIRECTORY);
```

Comment:
1. You can also inject `Config` with `@TestResource`.
2. The config key must be `ts.*`. The config system will ignore keys that don't begin with `ts.`.

Reply:
Changed